.github/workflows/cypress.yml

```diff
@@ -0,0 +1,87 @@
+name: Cypress Tests
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  # TODO: remove once speed/batch size testing complete
+  # interval of 6 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21 ]
+  # interval of 5 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 ]
+  # interval of 4 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ]
+  # interval of 3 files:
+  #       specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41 ]
+
+  e2e_tests:
+    
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21 ]
```

Comment:
FYI: Github Action supports dynamic matrices, so we can create a new job that will prepare these arrays or calculate them dynamically.
Here is an example: https://thekevinwang.com/2021/09/19/github-actions-dynamic-matrix/

This can be useful if you want to use a custom runner with more process or memory.
Pull requests from forks should not use custom runners for security reasons, so we would have to support two hardware configurations.